docs/OpenShift-Pipelines-CI.Next-proposal.md

```diff
@@ -0,0 +1,151 @@
+
+# Current release process overview
+
+* [openshift/tektoncd-pipeline](https://github.com/openshift/tektoncd-pipeline):
+
+  Release process document located here :
+  https://github.com/openshift/tektoncd-pipeline/blob/master/RELEASE_PROCESS.md
+
+  Steps are :
+
+  1) Create release branch and push to `openshift/tektoncd-pipeline`
+  2) Create a PR on `openshift/release` to add CI and image mirroring to quay
+     for the release branch.
+  3) Create a PR against the release branch in `openshift/tektoncd-pipeline`
+  with a newly generated `release.yaml`
+
+* [openshift/tektoncd-catalog](https://github.com/openshift/tektoncd-catalog):
+  Same steps as `openshift/tektoncd-pipeline` and located in the same [release process document](https://github.com/openshift/tektoncd-pipeline/blob/master/RELEASE_PROCESS.md).
+
+* [openshift/pipelines-catalog](https://github.com/openshift/pipelines-catalog):
+  Manual tagging/branch, no documention.
+
+* [openshift/tektoncd-cli](https://github.com/openshift/tektoncd-cli):
+  No downstream release, we are using upstream CLI.
+
+* [openshift/tektoncd-triggers](https://github.com/openshift/tektoncd-triggers):
+  Manual tagging/branch, no documention.
+
+* [openshift/tektoncd-pipeline-operator](https://github.com/openshift/tektoncd-pipeline-operator)
+  TBF: Documentation is [here](https://github.com/openshift/tektoncd-pipeline-operator/blob/master/docs/release.md)
+
+## OpenShift Pipelines CI.next
+
+### Requirements
+
+  * Monorepo CI with a top to bottom approach, if pipelines fails all the other fails.
+  * Tekton based.
+  * Run from the [devint](https://github.com/openshift-knative/cluster-devint) cluster.
+  * Run quietly, err verbosely. (less noise).
+  * Store artifacts. (ie `release.yaml`).
+  * Support credential install.
+
+### Optional / crazy long shot
+
+   * Overview Dashboard of all runs.
+   * Can be run from developer laptop.
+   * CI runs Metrics with performances and slowness detection.
+   * Error logs detection.
+
+### Implementation
+
+   * task1:
+       - using a buildah image
+
+       - checkout `openshift-pipelines/plumbing` as tekton git resource
+
+       - generate a container image from a Dockerfile with :
+         - Binaries: `oc`, `openshift4-install` `git`, `go`, `make`
+         - Probably other utilities and other tools for building application.
+         - Add `plumbing` repository somewhere into the image
+         - Not necessary but that would make things easier further down.
+
+       -  push it to the internal openshift registry.
+
+   * task2:
+       - run after task1
+
+       - using generated image from previous step.
+
+       - checkout `openshift-pipelines/plumbing` repo as tekton git resources.
+
+       - start and wait for an openshift4 temporary cluster install using an
```

Comment:
Can't we do that in parallel of other things ? (like compiling, linting, unit tests, …)

Reply:
ah I see, you want to execute everything on the temporary cluster instead of in `cluster-devint`

Reply:
That's correct, and that's unfortunate :( We want to be able to test against multiple openshift target....